spython/main/parse/parsers/base.py

```diff
@@ -0,0 +1,95 @@
+
+# Copyright (C) 2017-2019 Vanessa Sochat.
+
+# This Source Code Form is subject to the terms of the
+# Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+# with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+
+from spython.logger import bot
+from spython.utils import read_file
+from ..recipe import Recipe
+
+class ParserBase(object):
+    '''a parser Base is intended to provide helper functions for a parser,
+       namely to read lines in files, and otherwise interact with outputs.
+       Input should be some recipe (text file to describe a container build)
+       and output of parse() is a Recipe (spython.main.parse.recipe.Recipe)
+       object, which can be used to write to file, etc.
+    '''
+
+    lines = []
```

Comment:
Pretty sure you don't want a class variable here. Set this in `__init__`

Reply:
What's the difference?